qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py

```diff
@@ -86,66 +79,51 @@ def __init__(
         self,
         duration: float,
         offsets: np.ndarray,
-        rabi_rotations: Optional[np.ndarray] = None,
-        azimuthal_angles: Optional[np.ndarray] = None,
-        detuning_rotations: Optional[np.ndarray] = None,
-        name: str = None,
+        rabi_rotations: np.ndarray,
+        azimuthal_angles: np.ndarray,
+        detuning_rotations: np.ndarray,
+        name: Optional[str] = None,
     ):
 
         check_arguments(
             duration > 0,
             "Sequence duration must be above zero.",
             {"duration": duration},
         )
+
         offsets = np.asarray(offsets)
         check_arguments(
-            len(offsets) <= UPPER_BOUND_OFFSETS,
-            "Number of offsets is above the allowed number of maximum offsets. ",
-            {"offsets": offsets},
-            extras={
-                "number_of_offsets": len(offsets),
-                "allowed_maximum_offsets": UPPER_BOUND_OFFSETS,
-            },
-        )
-        check_arguments(
-            np.all(offsets >= 0.0) and np.all(offsets <= duration),
+            np.all((offsets >= 0) & (duration >= offsets)),
```

Comment:
```suggestion
            np.all(0 <= offsets <= duration),
```
Does that work?

Reply:
nope, python would complain `The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`,
and I was quite surprised.

Reply:
Ah, it's equivalent to `0 <= offsets and offsets <= duration` (https://docs.python.org/2.3/ref/comparisons.html), which would explain it. Oh well.